src/storage/slotted_page.c

```diff
@@ -1036,6 +1036,11 @@ spage_collect_statistics (PAGE_PTR page_p, int *npages, int *nrecords, int *rec_
 	  continue;
 	}
 
+      if (slot_p->record_type == REC_MARKDELETED || slot_p->record_type == REC_DELETED_WILL_REUSE)
+	{
+	  continue;
+	}
+
```

Comment:
How about this and removing `if (slot_p->offset_to_record == SPAGE_EMPTY_OFFSET)`?
```suggestion
      if (slot_p->record_type == REC_MARKDELETED || slot_p->record_type == REC_DELETED_WILL_REUSE)
	{
	  assert (slot_p->offset_to_record == SPAGE_EMPTY_OFFSET);
	  continue;
	}

```

Reply:
> the spage_mark_deleted_slot_as_reusable function does not set slot_p->offset_to_record to SPAGE_EMPTY_OFFSET

Even `spage_mark_deleted_slot_as_reusable()` requires that `slot_p->offset_to_record` is set to `SPAGE_EMPTY_OFFSET`.

Reply:
It's in terms of one of the reasons, readability. I don't understand `quick fix`. What is fixed?

Reply:
I called it a quick fix because it applied only the if condition necessary for the blocked issue. I think there is no problem on my fix when I checked it in the limited time. However I think I doesn't consider all cases enough.

Reply:
About spage_mark_deleted_slot_as_reusable() (re-visit)

The slot_p->offset_to_record should be the SPAGE_EMPTY_OFFSET when this function is called. So, I think it is no need to set the slot_p->offset_to_record to the SPAGE_EMPTY_OFFSET again. The purpose of this function is to change the value of the slot_p->record_type to REC_DELETED_WILL_REUSE.